File: 统计工具/affiliate-main/check-accounts.js
// 查询平台账号的affiliate_name字段
const Database = require('better-sqlite3');
const path = require('path');

const db = new Database(path.join(__dirname, 'data.db'));

console.log('\n=== 📋 平台账号列表 ===\n');

const accounts = db.prepare(`
  SELECT id, user_id, platform, account_name, affiliate_name, created_at
  FROM platform_accounts
  ORDER BY id
`).all();

if (accounts.length === 0) {
  console.log('暂无平台账号');
} else {
  console.log(`共 ${accounts.length} 个平台账号:\n`);

  accounts.forEach((acc, index) => {
    console.log(`${index + 1}. [ID: ${acc.id}] ${acc.platform} - ${acc.account_name}`);
    console.log(`   联盟序号: ${acc.affiliate_name || '(未设置)'}`);
    console.log(`   用户ID: ${acc.user_id}`);
    console.log(`   创建时间: ${acc.created_at}`);
    console.log('');
  });

  // 统计
  const withAffiliate = accounts.filter(a => a.affiliate_name).length;
  const withoutAffiliate = accounts.length - withAffiliate;

  console.log('📊 统计:');
  console.log(`   ✅ 已设置联盟序号: ${withAffiliate} 个`);
  console.log(`   ⚠️  未设置联盟序号: ${withoutAffiliate} 个`);
}

db.close();
console.log('\n=== ✅ 查询完成 ===\n');

File: backend/scripts/_check_accounts.py
import paramiko, sys
sys.stdout.reconfigure(encoding='utf-8')

ssh = paramiko.SSHClient()
ssh.set_missing_host_key_policy(paramiko.AutoAddPolicy())
ssh.connect("47.239.193.33", username="admin", password="A123456", timeout=10)

sftp = ssh.open_sftp()
with sftp.open("/tmp/check_accounts.py", "w") as f:
    f.write("""
import sys, os
sys.path.insert(0, "/home/admin/Google-Data-Analysis/backend")
os.chdir("/home/admin/Google-Data-Analysis/backend")

from app.database import SessionLocal
from sqlalchemy import text

db = SessionLocal()

# 1. affiliate_platforms schema
print("=== affiliate_platforms schema ===")
rows = db.execute(text("PRAGMA table_info(affiliate_platforms)")).fetchall()
for r in rows:
    print(f"  {r[1]} ({r[2]})")

# 2. All platforms
print()
print("=== Platforms ===")
rows = db.execute(text("SELECT * FROM affiliate_platforms")).fetchall()
for r in rows:
    print(f"  {r}")

# 3. affiliate_accounts schema
print()
print("=== affiliate_accounts schema ===")
rows = db.execute(text("PRAGMA table_info(affiliate_accounts)")).fetchall()
for r in rows:
    print(f"  {r[1]} ({r[2]})")

# 4. Active accounts with platform name
print()
print("=== Active Accounts ===")
rows = db.execute(text(
    "SELECT aa.user_id, u.username, ap.name as platform_name "
    "FROM affiliate_accounts aa "
    "JOIN users u ON u.id = aa.user_id "
    "JOIN affiliate_platforms ap ON ap.id = aa.platform_id "
    "WHERE aa.status = 'active' "
    "ORDER BY aa.user_id, ap.name"
)).fetchall()
current_uid = None
for r in rows:
    if r[0] != current_uid:
        current_uid = r[0]
        print(f"  {r[1]} (id={r[0]}):")
    print(f"    {r[2]}")

# 5. Platform config
print()
print("=== Platform Config ===")
try:
    from app.services.merchant_platform_sync import get_platform_config
    for pname in ['CG', 'BSH', 'LH', 'LB', 'PH', 'AW', 'RA', 'PM', 'RW']:
        try:
            cfg = get_platform_config(pname)
            mode = cfg.get('mode', '?')
            ms = cfg.get('max_size', '?')
            print(f"  {pname}: mode={mode}, max_size={ms}")
        except Exception as e:
            print(f"  {pname}: {e}")
except Exception as e:
    print(f"  Error: {e}")

db.close()
""")
sftp.close()

stdin, stdout, stderr = ssh.exec_command(
    "cd /home/admin/Google-Data-Analysis/backend && source venv/bin/activate && python3 /tmp/check_accounts.py 2>&1",
    timeout=30
)
print(stdout.read().decode('utf-8', errors='replace'), flush=True)

ssh.close()
